### PR TITLE
Add Team Drive Support

### PR DESCRIPTION
**Pull request by rtkg12**
Added Team Drive Support. Tested and working on Mac OS 10.14 and Ubuntu 16.04. 

Need to specify --root-node-id for the team drive. If not specified, it works normally with the root folder of the drive.

### CODE REVIEW

File: drive/drive.go [hunk 1]
```diff
@@ -105,7 +105,14 @@ func (d *Client) checkChanges(firstCheck bool) {
 		query := client.Changes.
 			List(pageToken).
 			Fields(googleapi.Field(fmt.Sprintf("nextPageToken, newStartPageToken, changes(removed, fileId, file(%v))", Fields))).
-			PageSize(1000)
+			PageSize(1000).
+			//TeamDriveId(d.rootNodeID).
```
Comment (felixbuenemann) [inline drive/drive.go]: Please remove commented code.

File: drive/drive.go [hunk 2]
```diff
@@ -105,7 +105,14 @@ func (d *Client) checkChanges(firstCheck bool) {
...
+			SupportsTeamDrives(true).
+			IncludeTeamDriveItems(true)
+
+		if d.rootNodeID != "root"{
```
Comment (felixbuenemann) [inline drive/drive.go]: Missing space before opening curly brace.

**Comment by felixbuenemann** [comment]
Sorry for the sluggish feedback, I'm very busy this week. I'll hopefully habe some time to test and merge this on Sunday. 

I looked at the CI test failures on Go 1.7 and they don't seem to be related to your changes.

**Comment by felixbuenemann** [comment]
@rtkg12 Can you descript how to get the `--root-node-id` for the teamdrive?

**Comment by felixbuenemann** [comment]
Please rebase on current master to fix the CI build.

**Comment by placid09** [comment]
Any update on these changes? Team Drive support would be a game changer IMO.

**Comment by zachgoold** [comment]
Would like to see an update on this if possible!

**Comment by Arudarin** [comment]
So... Is this greenlighted?

**Comment by felixbuenemann** [comment]
It needs to be rebased and it lacks information on how to find the correct root node id, so I couldn't verify it actually works.

**Comment by herver** [comment]
Hi there,

I managed to get this PR working on my setup.

In order to get the `Root Node ID` of a Team drive, you just open the Team Drive in your browser and you take the last part of the URL, example:

* URL: https://drive.google.com/drive/u/0/folders/ABC123qwerty987
* Root Node ID: `ABC123qwerty987`

Also the CI fails on Go 1.7 because the code uses `GetBody` which has been introduced only in Go 1.8 so a fix for this is needed as well (as Go < 1.10 is unsupported as per the Release policy)

**Comment by felixbuenemann** [comment]
@herver Thanks for testing, please open a new PR for the documentation changes.

**Comment by s0mm3rb** [comment]
hi there
awesome that teamdrive is added but I can't get it to work
I only get `cannot obtain root node: Could not get root object`
this is my mount command
`root@us-iowa-1:/# /usr/lib/plexdrive/plexdrive mount -v 2 /mnt/plexdrive --root-node-id=<removed>`
Full Debug:
`[USR/LIB/PLEXDRIVE/PLEXDRIVE] [2019-07-16 01:41] DEBUG  : Opening cache connection
[USR/LIB/PLEXDRIVE/PLEXDRIVE] [2019-07-16 01:41] DEBUG  : Authorizing against Google Drive API
[USR/LIB/PLEXDRIVE/PLEXDRIVE] [2019-07-16 01:41] DEBUG  : Loading token from cache
[USR/LIB/PLEXDRIVE/PLEXDRIVE] [2019-07-16 01:41] INFO   : Mounting path /mnt/plexdrive
[USR/LIB/PLEXDRIVE/PLEXDRIVE] [2019-07-16 01:41] DEBUG  : Checking for changes
[USR/LIB/PLEXDRIVE/PLEXDRIVE] [2019-07-16 01:41] DEBUG  : Getting start page token from cache
[USR/LIB/PLEXDRIVE/PLEXDRIVE] [2019-07-16 01:41] DEBUG  : Last change id found, continuing getting changes (60834)
[USR/LIB/PLEXDRIVE/PLEXDRIVE] [2019-07-16 01:41] INFO   : First cache build process started...
[USR/LIB/PLEXDRIVE/PLEXDRIVE] [2019-07-16 01:41] DEBUG  : Getting root from API
[USR/LIB/PLEXDRIVE/PLEXDRIVE] [2019-07-16 01:41] DEBUG  : Storing page token 60834 in cache
[USR/LIB/PLEXDRIVE/PLEXDRIVE] [2019-07-16 01:41] INFO   : First cache build process finished!
[USR/LIB/PLEXDRIVE/PLEXDRIVE] [2019-07-16 01:41] DEBUG  : googleapi: Error 404: File not found: <removed>., notFound
[USR/LIB/PLEXDRIVE/PLEXDRIVE] [2019-07-16 01:41] WARNING: Could not get object <removed> from API
[USR/LIB/PLEXDRIVE/PLEXDRIVE] [2019-07-16 01:41] DEBUG  : cannot obtain root node: Could not get root object`

thanks

**Comment by hardlick** [comment]
Hi, can you release ?

**Pull request merged**